runtime/lua/vim/lsp/handlers.lua

```diff
@@ -333,25 +335,41 @@ M['textDocument/implementation'] = location_handler
 ---     - border:     (default=nil)
 ---         - Add borders to the floating window
 ---         - See |vim.api.nvim_open_win()|
+---     - parametersOnly: (default=false)
+---         - Display only a list of parameters
+---     - suppressPrint: (default=false)
+---         - Stop printing a message if no signature is available
 function M.signature_help(_, method, result, _, bufnr, config)
   config = config or {}
   -- When use `autocmd CompleteDone <silent><buffer> lua vim.lsp.buf.signature_help()` to call signatureHelp handler
   -- If the completion item doesn't have signatures It will make noise. Change to use `print` that can use `<silent>` to ignore
   if not (result and result.signatures and result.signatures[1]) then
-    print('No signature help available')
+    if not config.suppressPrint then
+      print('No signature help available')
+    end
     return
   end
-  local lines = util.convert_signature_help_to_markdown_lines(result)
+  local lines, highlights = util.get_signature_help_with_highlights(result, config.parametersOnly or false)
   lines = util.trim_empty_lines(lines)
   if vim.tbl_isempty(lines) then
-    print('No signature help available')
+    if not config.suppressPrint then
+      print('No signature help available')
+    end
     return
   end
   local syntax = api.nvim_buf_get_option(bufnr, 'syntax')
+  if syntax == "" then
```

Comment:
I'm still not sure why syntax is null for you, it's not for me on go files, and I'd want to understand why before merging this change.

Reply:
No idea. I just know that it is, and I've never actually seen highlighted signature popups

Reply:
Something may be wrong with your config then: 

![image](https://user-images.githubusercontent.com/13316262/116788844-c4c8ac00-aa60-11eb-8f57-3bd7ffcd92b7.png)

Reply:
That's the most likely explanation, or some plugin. `verbose set syntax?` just returns `Last set from Lua`. Though this is the first time I've had a problem with the lack of a set `syntax` option.

Reply:
I have also been experiencing something similar (see also [https://github.com/ray-x/lsp_signature.nvim/issues/13](https://github.com/ray-x/lsp_signature.nvim/issues/13)). Turns out, if I enable treesitter highlighting, I lose the markdown highlighting in the signature help.
`init.lua`
```lua
vim.cmd('packadd! nvim-lsp')
vim.cmd('packadd! nvim-treesitter')
vim.api.nvim_set_keymap("n", "gk", "<cmd>lua vim.lsp.buf.signature_help()<CR>", {})
require'lspconfig'.gopls.setup{}
require'nvim-treesitter.configs'.setup {
    highlight = {
        enable = true,
    },
}
```
`nvim -u init.lua test.go`
![2021-05-02-122439_859x195_scrot](https://user-images.githubusercontent.com/32982619/116809973-61fb1180-ab41-11eb-97b1-9a1a02f3baa6.png)
`init_.lua`
```lua
vim.cmd('packadd! nvim-lsp')
vim.api.nvim_set_keymap("n", "gk", "<cmd>lua vim.lsp.buf.signature_help()<CR>", {})
require'lspconfig'.gopls.setup{}
```
`nvim -u init_.lua test.go`
![2021-05-02-122722_837x165_scrot](https://user-images.githubusercontent.com/32982619/116810043-c28a4e80-ab41-11eb-8780-52277d808e84.png)
Not sure what the problem is here, of if this is intendet behavior. But I would be interested in how this is resolved.

Reply:
I turn it off syntax in my init.lua because I am using treesitter.
That can save 15ms~200ms when loading.   As syntax highlighting can be slow and if you need treesitter to render it twice, makes loading slower.
I can use treesitter to highlight the signature help. The only problem is it is missing `func` before Println and treesitter will complaining about it.